### PR TITLE
Implement toMap() with a target. Fixes issue #656

### PR DESCRIPTION
**Pull request by cguntur-bnym**

### CODE REVIEW

File: eclipse-collections-code-generator/src/main/resources/impl/map/immutable/immutablePrimitiveObjectEmptyMap.stg [hunk 1]
```diff
@@ -568,6 +569,12 @@ final class Immutable<name>ObjectEmptyMap\<V> implements Immutable<name>ObjectMa
         return Maps.mutable.with();
     }
 
+    @Override
+    public \<NK, NV, R extends Map\<NK, NV>\> R toMap(Function\<? super V, ? extends NK> keyFunction, Function\<? super V, ? extends NV> valueFunction, R target)
+    {
+        throw new UnsupportedOperationException("Cannot call toMap with a generic Map target on " + this.getClass().getSimpleName());
```
Comment (donraab) [inline eclipse-collections-code-generator/src/main/resources/impl/map/immutable/immutablePrimitiveObjectEmptyMap.stg]: This should return an empty Map.

File: eclipse-collections-code-generator/src/main/resources/impl/map/immutable/immutablePrimitiveObjectSingletonMap.stg [hunk 2]
```diff
@@ -681,6 +682,12 @@ final class Immutable<name>ObjectSingletonMap\<V> extends AbstractImmutable<name
         return Maps.mutable.with(keyFunction.valueOf(this.value1), valueFunction.valueOf(this.value1));
     }
 
+    @Override
+    public \<NK, NV, R extends Map\<NK, NV>\> R toMap(Function\<? super V, ? extends NK> keyFunction, Function\<? super V, ? extends NV> valueFunction, R target)
+    {
+        return target;
```
Comment (donraab) [inline eclipse-collections-code-generator/src/main/resources/impl/map/immutable/immutablePrimitiveObjectSingletonMap.stg]: This should either be deleted or call target.put() with the single key and value first.

**Comment by cguntur-bnym** [comment]
Hi Don, Thank you for the review and feedback. I will add the necessary missing tests today.

________________________________

**Comment by cguntur-bnym** [comment]
Additionally includes fixes for https://github.com/eclipse/eclipse-collections/issues/704

**Pull request merged**